pages/positions/addliquidity.tsx

```diff
@@ -22,51 +13,33 @@ const style = {
   tokenInput: "bg-transparent placeholder:text-gray-600 outline-none mb-6 w-full text-4xl text-gray-300 mt-4",
   nonSelectedTokenContainer: "flex items-center w-1/2 text-gray-200",
   nonSelectedTokenContent: "w-full h-min flex justify-center items-center bg-blue-500 rounded-2xl text-2xl font-medium cursor-pointer p-2 mt-[-0.2rem] shadow-lg shadow-blue-500/30 hover:bg-blue-600 hover:shadow-blue-600/30",
-  tokenSelectorContainer: "flex items-center w-1/4 text-gray-200",
+  tokenSelectorContainer: "flex flex-col items-center w-1/3 text-gray-200",
   tokenSelectorContent: "w-full h-min flex justify-between items-center bg-gray-700 rounded-2xl text-xl font-medium cursor-pointer p-2 mt-[-0.2rem] shadow-lg shadow-gray-700/30 hover:bg-gray-900 hover:shadow-gray-900/30",
   tokenSelectorIcon: "flex items-center",
   tokenSelectorTicker: "mx-2",
+  tokenBalance: "self-end mt-2 text-sm text-gray-300 opacity-50",
   dropdownArrow: "w-12 h-8",
   invalidatedButton: "disabled my-2 rounded-2xl py-4 px-6 text-xl font-semibold flex justify-center items-center text-gray-600 mt-8 border-2 border-gray-700",
   confirmButton: "bg-blue-400 my-2 rounded-2xl py-4 px-6 text-xl font-semibold flex justify-center items-center cursor-pointer text-white mt-8 border-2 border-blue-400 hover:border-blue-300"
 }
 
 const AddLiquidity: NextPage = () => {
-  // holds state of user amount inputted for each of the token input fields
-  const [tokenAInputVal, setTokenAInputVal] = useState<string>("")
-  const [tokenBInputVal, setTokenBInputVal] = useState<string>("")
-
-  // holds state of token selected for both token A and B input fields
-  const [tokenASelected, setTokenASelected] = useState<Token>(Tokens[0])
-  const [tokenBSelected, setTokenBSelected] = useState<Token>({
-    imgPath: "",
-    symbol: "",
-    address: "",
-  })
-
-  // holds state of what token input field was selected
-  const [tokenSelected, setTokenSelected] = useState<string>("")
-
-  // holds state of modal open and close
-  const [isOpen, setIsOpen] = useState<boolean>(false)
-
-  // holds global state of user info and ethers provider for contract calls
-  const { accountInfo, provider } = useContext(WalletContext)
-  
-  // checks for non-numeric value inputs
-  const validateTokenInput = (
-    e: ChangeEvent<HTMLInputElement>,
-    setToken: Dispatch<SetStateAction<string>>
-  ): void => {
-    const tokenInput: string = (e.target as HTMLInputElement).value
-    setToken(tokenInput.replace(/[^0-9.]/g, ''))
-  }
-
-  // checks which tokenSelector element was selected and opens modal
-  const handleTokenSelector = (tokenSelected: string): void => {
-    setTokenSelected(tokenSelected)
-    setIsOpen(true)
-  }
+  const { 
+    tokenAInputVal,
```

Comment:
So it looks like all the values and handlers are in the handlers file as another way of being a controller. Is there a name for this architectural style with guidance and guidelines to follow?

Reply:
Yeah, this is React's form of the MVC architechture. Per [react's docs](https://reactjs.org/docs/hooks-custom.html), it's a `.ts` file that handles all of the logic that the normal component file would of had (all code before the `return`). Usually, React suggests to name these custom hook files prefixed with `use`, to identify it. 

The way you and Usama would do it from now on is like this. Since React does not reflect the MVC pattern (the view is already the component itself), you would not need to separate it as MVC. You can simply build the component (view) and custom hook as the logic (controller, and better if it can be reused in other components). What is great for us is that we don't need a Model, since the backend is decentralized and autonomous. 

**TL;DR** - guideline for creating any sort of component with logic is by having a normal component for UI and extracted logic into a custom hook prefixed by `use`

Reply:
But this is for react state interaction I think. Is it suitable for async nature of web3?

Reply:
Seems reasonable to have react hooks. Does this mean that all the hooks will be encoded in the same page? As the page is seems kind of bloated already.